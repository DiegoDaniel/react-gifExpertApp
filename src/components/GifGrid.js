import React from 'react' //, { useEffect, useState }
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( { category } ) => {
            //renombro
    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>
            { loading && <p className="animate__animated animate__flash">Cargando...</p>}
            <div className="card-grid">
                { 
                images.map( ( img ) => {
                    return <GifGridItem 
                        key={ img.id }
                        //Se puede hacer spread de CADA UNO de los valores de img
                        //Para enviar al componente, en lugar de hacer id={img.id},, title={img.title},
                        //El spread devuelve todos los valores del objeto.
                        { ...img } />;
                })
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}