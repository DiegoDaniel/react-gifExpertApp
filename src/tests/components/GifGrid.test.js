import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en GifGrid', () => {
    
    const category = 'Goku';
    
    test('should match snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar items cuando se carga el useFetch', () => {
        
        const gifs = [{
            id: '123',
            url: 'https://localhost/todo.png',
            title: 'Local Host'
        },{
            id: 'abc',
            url: 'https://localhost/todo.png',
            title: 'Local Host'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category}/>);

        //expect( wrapper ).toMatchSnapshot();
        //No deber√≠a existir el <p></p> porque "loading" es false.
        expect( wrapper.find('p').exists()).toBe( false );

        //Tambien podemos esperar que se haya renderizado el componente GifGridItem
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
    
    
})
