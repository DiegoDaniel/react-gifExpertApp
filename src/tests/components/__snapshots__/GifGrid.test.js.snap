// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid should match snapshot 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
